Makefile

```diff
@@ -0,0 +1,23 @@
+# See: https://gist.github.com/asukakenji/f15ba7e588ac42795f421b48b8aede63
+# For a list of valid GOOS and GOARCH values
+# Note: these can be overriden on the command line e.g. `make PLATFORM=<platform> ARCH=<arch>`
+PLATFORM="$(shell go env GOOS)"
+ARCH="$(shell go env GOARCH)"
+
+
+all: agent docker-credential-portainer download-binaries
+
+agent:
+	@echo "Building agent..."
+	@GOOS=$(PLATFORM) GOARCH=$(ARCH) CGO_ENABLED=0 go build --installsuffix cgo --ldflags '-s'  -o dist/$@ cmd/agent/main.go
+
+docker-credential-portainer:
+	@echo "Building docker-credential-portainer..."
+	@cd cmd/docker-credential-portainer; \
+	GOOS=$(PLATFORM) GOARCH=$(ARCH) CGO_ENABLED=0 go build --installsuffix cgo --ldflags '-s' -o ../../dist/$@
+
+download-binaries:
```

Comment:
when do you use this one? only independently ? I thought it was required by the agent install?